src/lib/types/events.ts

```diff
@@ -135,9 +136,15 @@ class BaseEvent implements IBaseEvent {
 
     readonly tags: ITag[];
 
-    constructor(type: string, createdBy: string, tags: ITag[] = []) {
+    /**
+     * @param createdBy accepts a string for backward compatibility. Prefer using IUser for standardization
+     */
+    constructor(type: string, createdBy: string | IUser, tags: ITag[] = []) {
         this.type = type;
-        this.createdBy = createdBy;
+        this.createdBy =
+            typeof createdBy === 'string' || typeof createdBy === 'undefined'
+                ? createdBy
```

Comment:
This maintains the same behaviour as before, if `createdBy` is passed as undefined we propagate that (which should break DB constraints and display an error)